pkg/kubectl/cmd/edit.go

```diff
@@ -0,0 +1,419 @@
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/editor"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/jsonmerge"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	//	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	//	"github.com/openshift/origin/pkg/cmd/util/editor"
+	//"github.com/openshift/origin/pkg/util/jsonmerge"
+)
+
+const (
+	editLong = `Edit a resource from the default editor.
+
+The edit command allows you to directly edit any API resource you can retrieve via the
+command line tools. It will open the editor defined by your OSC_EDITOR, GIT_EDITOR,
```

Comment:
I think it's a good idea to use the existing env variable. I'll take @smarterclayton 's suggestion.